### ISSUE TITLE
[Bug]: FC0002 - False positive for XmlPort

### ISSUE THREAD

**Issue by pri-kise** [ALCops/Analyzers#432]
### Analyzer

FormattingCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

FC0002

### Description

Wrong warning for `Use 'Xmlport' instead of 'XmlPort' for consistent casing.`

### AL code to reproduce

```al
            action(Import)
            {
                Caption = 'Import';
                Image = Import;
                ToolTip = 'Import asdf.';

                trigger OnAction()
                begin
                    XmlPort.Run(XmlPort::"PTE Import");
                end;
            }
```

### Additional context

https://github.com/StefanMaron/BusinessCentral.LinterCop/issues/729

**Comment by Arthurvdv** [comment]
Thanks for the report! I dug into this one, because the XmlPort casing keeps coming up (see also StefanMaron/BusinessCentral.LinterCop#729). After checking the AL compiler internals, I believe the diagnostic is actually **correct here**, and I'd like to verify that assumption with you.

## Why FC0002 suggests `Xmlport` in this context

AL itself is intentionally inconsistent about the casing of "xmlport" — the canonical spelling depends on the context:

| Context | Canonical casing | Source |
|---|---|---|
| Object declaration | `xmlport 50100 "My Xmlport"` | Keyword text in the compiler (`SyntaxFacts`) |
| Variable / parameter / return type | `MyPort: XmlPort "My Xmlport";` | `NavTypeKind.XmlPort` display name |
| Static class access (`Run`, `Import`, `Export`) | `Xmlport.Run(...)` | The compiler's built-in class symbol is literally named `Xmlport` |
| Object access (left of `::`) | `Xmlport::"My Xmlport"` | Binds to that same `Xmlport` class symbol |
| `ObjectType` option member | `ObjectType::XmlPort` | `SymbolKind` enum name |

In your snippet, both usages fall into the `Xmlport` contexts, so the canonical form is:

```al
trigger OnAction()
begin
    Xmlport.Run(Xmlport::"PTE Import");
end;
```

This also matches what IntelliSense from the official AL Language extension suggests:

<img width="774" height="336" alt="Image" src="https://github.com/user-attachments/assets/54d2c897-cdc3-42f9-92a1-973f1069af3e" />

## Can you double-check?

Could you verify what IntelliSense suggests on your end when you type `Xml...` inside the `Run(...)` call and before `.Run`? If it also proposes `Xmlport`, I'd treat this diagnostic as working as intended and close the issue. If your IntelliSense proposes `XmlPort` there, please share a screenshot and your AL Language extension version, because then we have a real discrepancy to look at.

I know the context-dependent casing is annoying (you write `XmlPort` as a variable type and get told to write `Xmlport` two lines later), but FC0002 deliberately follows the compiler's own canonical names rather than inventing a single spelling.

We've also updated the FC0002 documentation on alcops.dev with a section clarifying this: [XmlPort vs Xmlport casing](https://alcops.dev/docs/analyzers/formattingcop/fc0002/#xmlport-vs-xmlport-casing). Regression tests locking down each context have been added as well.

---
*Generated with Claude Fable 5*

**Comment by Arthurvdv** [comment]
@pri-kise did you happen to have a chance to take a look at my feedback?

**Comment by pri-kise** [comment]
well I guess. that I can live it it.
I'm now only waiting for the day when a new AL Language will fix the inconsistency and I will have to update the code.
But with AI this is okay.

**Comment by Arthurvdv** [comment]
I understand the confusion with the inconsistent casing, it's one of those quirks of the AL Language I believe.

Regarding the possibility of a future AL Language update fixing the inconsistency: the casing topic has been discussed at length on the BusinessCentral.LinterCop side as well, and I don't expect it to change anytime soon (but of course there are no guarantees). If Microsoft does unify the casing in a future compiler update, we'll update FC0002 accordingly.

So if it would it ever change, there's a QuickFix available in VS Code that lets you update the casing with a single click, so it should be easy to fix across your codebase.

<img width="586" height="290" alt="Image" src="https://github.com/user-attachments/assets/a96a2506-b69c-4620-ba5b-69aaa7a8571e" />
